### ISSUE TITLE
codegen does not work locally

### ISSUE THREAD

**Issue by ideepika** [rook/rook#15047]
```
ook git:(wip-jaeger-2024) ✗ make codegen
mkdir -p /Users/deepika/rook2/rook/.cache/tools/linux_arm64
curl -sL https://github.com/kubernetes/code-generator/archive/refs/tags/v0.20.0.tar.gz | tar -xz -C /Users/deepika/rook2/rook/.cache/tools/linux_arm64
Generating deepcopy funcs
F1125 00:42:47.116734   23974 main.go:82] Error: Failed executing generator: some packages had errors:
errors in package "github.com/rook/rook/pkg/apis/ceph.rook.io/v1":
open ../../../github.com/rook/rook/pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go: no such file or directory

goroutine 1 [running]:
k8s.io/klog/v2.stacks(0x1)
	/Users/deepika/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1026 +0x8c
k8s.io/klog/v2.(*loggingT).output(0x1053d12a0, 0x3, {0x0, 0x0}, 0x1400fd92a10, {0x10525be88?, 0x1?}, 0x14000108900?, 0x0)
	/Users/deepika/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:975 +0x538
k8s.io/klog/v2.(*loggingT).printf(0x140000e0e60?, 0xe7de8?, {0x0, 0x0}, {0x0, 0x0}, {0x1050d1b1c, 0x9}, {0x1401563d760, 0x1, ...})
	/Users/deepika/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:750 +0x180
k8s.io/klog/v2.Fatalf(...)
	/Users/deepika/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1502
main.main()
	/Users/deepika/rook2/rook/.cache/tools/linux_arm64/code-generator-0.20.0/cmd/deepcopy-gen/main.go:82 +0x430

goroutine 21 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x0?)
	/Users/deepika/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1169 +0x5c
created by k8s.io/klog/v2.init.0 in goroutine 1
	/Users/deepika/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:417 +0x118
make: *** [codegen] Error 255
```
maybe script needs some tweaking, but I am not sure why it's failing right now

**Comment by sp98** [comment]
Please update the issue with the environment being used.  Rook version, OS etc.

**Comment by sp98** [comment]
>Error: Failed executing generator: some packages had errors:

Try updating the dependencies in your local repository

**Comment by obnoxxx** [comment]
for me, it works locally  without error on Fedora Linux 41 and MacOS 15.1.1

**Comment by obnoxxx** [comment]
Also worked on Ubuntu 18.04 LTS for me after fixing the shebang line in the `codegen.sh` script

**Comment by obnoxxx** [comment]
trying once more on ubuntu 22.04, I get this error:

```console
$ make codegen
build/makelib/golang.mk:27: *** please set GO_STATIC_PACKAGES prior to including golang.mk.  Stop.
$ 

```

Thw same error occurs if I try any  other golab-related target, like `check.mod` or `go.vet` or `test`.

I know have seen this before but I don't remember how I worked around it. ...

**Comment by BlaineEXE** [comment]
Try setting `export GOARCH="$(go env GOARCH)"` in the shell?

**Comment by travisn** [comment]
@ideepika Perhaps it's related to the path not being under `github.com`, related to #14984.

**Comment by obnoxxx** [comment]
@ideepika , could you try if this helps: #15075 ? 

Just asking, ut from your logs, I don't  really think this is the cause.

**Comment by ideepika** [comment]
Hey @obnoxxx I will give it a try, but maybe get a bit delayed testing it, came for cephalacon.

**Comment by BlaineEXE** [comment]
I'm pretty confident this is resolved by #14915. If not, we can reopen
